tests/test_miscellaneous.py

```diff
@@ -44,3 +46,24 @@ def test_properties_setters(self):
     def test_asdict(self):
         dict_ = self.subject._asdict()
         assert_equal(dict_['script'], self.subject.script)
+
+
+# This does not subclass TestCase so test generators work
```

Comment:
[Here](http://nose.readthedocs.org/en/latest/writing_tests.html#test-generators). Basically whenever a test method is a [generator](https://wiki.python.org/moin/Generators), nose will treat each of the values it produces as instructions to construct a test case. The first element of each tuple is something to call, and the rest are arguments.

It differs from simply calling the check function multiple times in that each case is separate and will execute on it's own, instead of in a sequence where any failure would fail the whole test.